src/main/java/pro/belbix/ethparser/web3/Web3Functions.java

```diff
@@ -344,7 +381,10 @@ public double fetchAverageGasPrice(String network) {
     }
     EthFilter filter = new EthFilter(fromBlock,
         toBlock, addresses);
-    filter.addOptionalTopics(topics);
+    for (String topic : mandatoryTopics) {
+      filter.addSingleTopic(topic);
```

Comment:
web3j has two methods but both add in the same structure
I think "optional" and "mandatory" topics do the same things
check please web3j lib sources

Reply:
I've checked the behavior and added tests. There is a difference in mandatory vs optional topics.